code/modules/mob/living/simple_animal/guardian/guardian_abilities.dm

```diff
@@ -0,0 +1,733 @@
+			S.spawner = owner
+			S.name = "[get_area(snare_loc)] snare ([rand(1, 1000)])"
+			I.snares |= S
+			owner << "<span class='danger'><B>Surveillance snare deployed!</span></B>"
+		else
+			owner << "<span class='danger'><B>You have too many snares deployed. Remove some first.</span></B>"
+
+/datum/action/innate/snare/remove
+	name = "Remove Snare"
+	button_icon_state = "camera_off"
+
+/datum/action/innate/snare/remove_snare/Activate()
+	var/mob/living/simple_animal/hostile/guardian/A = owner
+	for(var/datum/guardian_abilities/ranged/I in A.current_abilities)
+		var/picked_snare = input(owner, "Pick which snare to remove", "Remove Snare") as null|anything in I.snares
+		if(picked_snare)
+			owner -= picked_snare
+			qdel(picked_snare)
+			owner << "<span class='danger'><B>Snare disarmed.</span></B>"
+
+
+
+/datum/guardian_abilities/ranged/ranged_attack(atom/target_atom)
+	if(istype(., /obj/item/projectile))
+		var/obj/item/projectile/P = .
+		if(stando.namedatum)
+			P.color = stando.namedatum.colour
+
+//magician's fried chicken
+
+/datum/guardian_abilities/fire/handle_stats()
+	stando.melee_damage_lower += 4.5
+	stando.melee_damage_upper += 4.5
+	stando.attack_sound = 'sound/items/Welder.ogg'
+	stando.damage_coeff -= list(BRUTE = 0.3, BURN = 0.3, TOX = 0.3, CLONE = 0.3, STAMINA = 0, OXY = 0.3)
+	stando.range += 4.5
+	stando.a_intent = INTENT_HELP
+
+/datum/guardian_abilities/fire/bump_reaction(AM as mob|obj)
+	if(isliving(AM))
+		var/mob/living/M = AM
+		if(!stando.hasmatchingsummoner(M) && M != user && M.fire_stacks < 7)
+			M.fire_stacks = 7
+			M.IgniteMob()
+
+/datum/guardian_abilities/fire/life_act()
+	. = ..()
+	if(user)
+		user.ExtinguishMob()
+		user.adjust_fire_stacks(-20)
+
+
+//lame
+
+/datum/guardian_abilities/assassin
+	var/stealthcooldown = 160
+	var/obj/screen/alert/canstealthalert
+	var/obj/screen/alert/instealthalert
+
+/datum/guardian_abilities/assassin/handle_stats()
+	stando.melee_damage_lower += 7
+	stando.melee_damage_upper += 7
+	stando.attacktext = "slashes"
+	stando.attack_sound = 'sound/weapons/bladeslice.ogg'
+	stando.toggle_button_type = /obj/screen/guardian/ToggleMode/Assassin
+
+
+/datum/guardian_abilities/assassin/proc/updatestealthalert()
+	if(stealthcooldown <= world.time)
+		if(toggle)
+			if(!instealthalert)
+				instealthalert = stando.throw_alert("instealth", /obj/screen/alert/instealth)
+				stando.clear_alert("canstealth")
+				canstealthalert = null
+		else
+			if(!canstealthalert)
+				canstealthalert = stando.throw_alert("canstealth", /obj/screen/alert/canstealth)
+				stando.clear_alert("instealth")
+				instealthalert = null
+	else
+		stando.clear_alert("instealth")
+		instealthalert = null
+		stando.clear_alert("canstealth")
+		canstealthalert = null
+
+
+/datum/guardian_abilities/assassin/life_act()
+	. = ..()
+	updatestealthalert()
+	if(stando.loc == user && toggle)
+		stando.ToggleMode(0)
+
+/datum/guardian_abilities/assassin/handle_mode(forced = 0)
```

Comment:
Multiple mode stands will absolutely be a problem. I've got no idea how you'd handle that gracefully.